src/Hooks/StrictEqualityHooks.php

```diff
@@ -109,15 +122,28 @@ private static function isEqualOrdered(Atomic $first_type, Atomic $second_type):
             return true;
         }
 
-        if ($first_type instanceof Atomic\TArray && $second_type instanceof Atomic\TArray) {
+        if ($first_type instanceof Atomic\TList && $second_type instanceof Atomic\TList) {
+            return true;
```

Comment:
Ideally, each element from the list should be compared to its counterpart on the other list. For example, if you have `'0'` in the first list and `0` the other, == and === won't have the same behaviour.

If you don't want to bother for now, we can just drop the list case